openrtb-json/src/main/scala/com/powerspace/openrtb/json/BidRequestSerde.scala

```diff
@@ -1,17 +1,114 @@
 package com.powerspace.openrtb.json
 
-import com.google.openrtb.BidRequest
-import io.circe.{Decoder, Encoder, Json}
+import com.google.openrtb.BidRequest.Imp.Banner.Format
+import com.google.openrtb.BidRequest.Imp.Pmp.Deal
+import com.google.openrtb.BidRequest.Imp.Video.CompanionAd
+import com.google.openrtb._
+import com.google.openrtb.BidRequest.Imp.{Audio, Banner, Metric, Native, Pmp, Video}
+import com.google.openrtb.BidRequest.{Imp, Source}
+import com.powerspace.openrtb.json.util.JsonUtils
+import io.circe.generic.extras.Configuration
+
+
 
 /**
   * Serialize and Deserialize an OpenRTB BidRequest
   */
 object BidRequestSerde {
+  import io.circe._
+  import io.circe.syntax._
+  import io.circe.generic.extras.semiauto._
+
+  import JsonUtils._
+  implicit val customConfig: Configuration = Configuration.default.withSnakeCaseMemberNames
+
+  implicit val bannerAdTypeEncoder: Encoder[BannerAdType] = protobufEnumEncoder[BannerAdType]
```

Comment:
I think it is better to use absolute path (ie: `Imp.banner.banneradtype`)

Reply:
These guys are enums and they're all at `openrtb` package level